### ISSUE TITLE
The teardown function is called twice when test is async and t.plan() is used

### ISSUE THREAD

**Issue by aral** [substack/tape#551]
## Summary

The teardown function is called twice when test is async and `t.plan()` is used. (Tape version: 5.2.0.)

## To reproduce

```js
import test from 'tape'

test ('something', async t => {
  t.plan(1)
  t.teardown(function () {
    console.log('Tearing down!')
  })
  t.pass()
})
```

## What should happen

The teardown function should be called just once.

## What actually happens

The teardown function is called twice.

## Notes

This issue _does not_ trigger when the test is synchronous or when `t.end()` is used.

**Comment by ljharb** [comment]
Wow, great find :-) will fix asap

**Comment by aral** [comment]
@ljharb Thanks, Jordan, appreciate it :)

**Comment by ljharb** [comment]
hmm - oddly enough in my test, i'm able to reproduce this example as calling it *zero* times (which is still a bug), but not twice. When I add your example as a lone file, it calls it twice. This is clearly a fun one.

**Comment by ljharb** [comment]
Released in v5.2.1.

**Comment by aral** [comment]
Thanks so much, Jordan. It’s working like a charm :)
